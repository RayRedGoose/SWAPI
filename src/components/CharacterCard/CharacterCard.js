import React from 'react';
import './CharacterCard.scss';
import FavoriteButton from '../FavoriteButton/FavoriteButton'

const CharacterCard = ({name, species, planet, population, films, addFavoriteCard}) => {

  const filmList = () => {
    films.map(film => {
      return <li>{film}</li>
    })
  }

  return (
    <section className="character-card">
      <header>
        <h3>{name}</h3>
        <p>{species}</p>
      </header>
      <section className="character-card-planet">
      <div>
        <p>{planet}:</p>
        <p>{population}</p>
      </div>
      <ul>
        <h4 className="character-movies-header">Movies:</h4>
        {filmList}
      </ul>
      </section>
      <footer>
        <FavoriteButton card={name, species, planet, population, films} addFavoriteCard={addFavoriteCard} /> 
      </footer>
    </section>
  )
}

export default CharacterCard;
