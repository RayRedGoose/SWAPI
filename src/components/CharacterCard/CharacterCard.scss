.character-card {
  background-color: rgba(255, 255, 255, 0.8);
  font-size: 18px;
  margin-bottom: 20px;
  margin-right: 17px;
  padding: 20px;
  position: relative;
  width: 20%;
  header {
    border-bottom: 2px solid #5A2C05;
    text-align: center;
    width: 80%;
    margin: auto;
    h3 {
      color: #5A2C05;
      font-size: 24px;
      min-height: 60px;
    }
    p {
      color: rgba(90, 44, 5, 0.5);
      text-transform: lowercase;
      padding: 10px;
    }
  }
  footer {
    text-align: center;
  }
}

.character-card-planet {
  div {
    display: flex;
    justify-content: space-between;
    padding: 20px 0;
    p {
      color: #5A2C05;
      text-transform: uppercase;
      &:first-child {
        margin-left: 25px;
        &::before {
          content: '';
          width: 20px; height: 20px;
          background-image: url('../../images/mars.svg');
          position: absolute;
          left: 20px;
        }
      }
    }
  }
  ul {
    h4 {
      color: #5A2C05;
      border-bottom: 2px solid #5A2C05;
      padding-bottom: 5px;
      width: 40%;
      margin-bottom: 20px;
    }
    li {
      list-style-type: none;
      margin-left: 40px;
      margin-bottom: 15px;
      &:before {
        content: '';
        width: 20px; height: 20px;
        background-image: url('../../images/st.svg');
        position: absolute;
        left: 30px;
      }
    }
  }
}
