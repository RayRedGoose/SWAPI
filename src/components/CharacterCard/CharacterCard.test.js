import React from 'react'
import characterResults from '../../mockData/fakeCharacters.js'
import { shallow } from 'enzyme'
import CharacterCard from './CharacterCard'

describe('CharacterCard', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(
      <CharacterCard
        key={'card1'}
        isFavorite={true}
        character={characterResults[0]}
        addFavorite={jest.fn()}
        removeFavorite={jest.fn()}
      />)
  })

  it('should render CharacterCard with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should update state isFavorite based on recieved prop after rendering', () => {
    expect(wrapper.state('isFavorite')).toEqual(true)
  })

  it('should toggle state isFavorite when toggleCard is called', () => {
    expect(wrapper.state('isFavorite')).toEqual(true)

    wrapper.instance().toggleCard()

    expect(wrapper.state('isFavorite')).toEqual(false)
  })
})
