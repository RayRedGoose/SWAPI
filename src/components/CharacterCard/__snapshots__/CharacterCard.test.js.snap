// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCard should render CharacterCard with all data passed in correctly 1`] = `
<section
  className="character-card"
>
  <FavoriteButton
    addFavorite={[MockFunction]}
    card={
      Object {
        "films": Array [
          "bambi",
          "little women",
          "sound of music",
        ],
        "isFavorite": false,
        "name": "Luke",
        "planet": "Tatioone",
        "population": 20000,
        "species": "human",
      }
    }
    isFavorite={true}
    removeFavorite={[MockFunction]}
    toggleCard={[Function]}
  />
  <header>
    <h3>
      Luke
    </h3>
    <p>
      human
    </p>
  </header>
  <div
    className="character-card-planet"
  >
    <p>
      Tatioone
    </p>
    <p>
      20000
    </p>
  </div>
  <ul
    className="character-card-movies"
  >
    <h4>
      Movies:
    </h4>
    <li
      key="movie1"
    >
      bambi
    </li>
    <li
      key="movie2"
    >
      little women
    </li>
    <li
      key="movie3"
    >
      sound of music
    </li>
  </ul>
</section>
`;
